lib/prelude.dx

```diff
@@ -862,6 +910,24 @@ instance {a b c} [Subset a b, Subset b c] Subset a c
     Nothing -> Nothing
     Just y -> project a y
 
+instance {a b} Subset a (a|b)
+  inject' = \x. Left x
+  project' = \x. case x of
+    Left  x -> Just x
+    Right x -> Nothing
+
+instance {a b} Subset a (b|a)
+  inject' = \x. Right x
+  project' = \x. case x of
+    Left  x -> Nothing
+    Right x -> Just x
+
+instance {a b} [NonEmpty b, Eq b] Subset a (a & b)
+  inject' = \x. (x, first_ix)
+  project' = \(x, y). case y == first_ix of
+    True  -> Just x
+    False -> Nothing
```

Comment:
This is an instance that's a bit too specific to include in the prelude IMO. It would be better to just write those exact injections where you need them.

Reply:
Oh, I guess you're saying that because the specific injection used here might not match what other users want to do?  I agree, will remove.

This is probably too fine-grained, but I kind of want to split `Subset` into an empty typeclass `IsSubset` and another one like `HasInjection` which contains these methods.  That would let us keep the general assertion in the prelude.  Anyways this is all probably premature without an application.